Content.Server/Power/EntitySystems/ExtensionCableSystem.cs

```diff
@@ -258,14 +260,18 @@ private bool TryFindAvailableProvider(EntityUid owner, float range, [NotNullWhen
 
                 if (!provider.Connectable) continue;
 
-                if ((Transform(entity).LocalPosition - xform.LocalPosition).Length > Math.Min(range, provider.TransferRange)) continue;
+                var distance = (Transform(entity).LocalPosition - xform.LocalPosition).Length;
```

Comment:
Use `LengthSquared` instead, it avoids having to take the square root. Since you're only doing sorting, that's sufficient.

Reply:
Unfortunately it's used both for validating the distance _and_ then used for sorting.
I tried `LengthSquared` and the tests passed and everything looked fine, but in-game testing showed I got this wrong

And as I'm not good enough at math I left it in for now, but will revisit it after I manage to write a test for the entire functionality

Reply:
I managed to get a working test for the range, but my naïve setup of just switching to `LengthSquared` and then wrapping the `Math.min` in `MathF.sqrt` didn't work so at this point I think I will leave it as it is 😅